ticketingApp/main.js

```diff
@@ -0,0 +1,156 @@
+import './style.css'
+import javascriptLogo from './javascript.svg'
+import viteLogo from '/vite.svg'
+import { setupCounter } from './counter.js'
+
+
+function renderEventTemplate(){
+  const mainContentDiv = document.querySelector('.main-content-component');
+  mainContentDiv.innerHTML = getEventsTemplate();
+  displayCards();
+}
+
+function getEventsTemplate(){
+  return `
+  <header>
+  <nav>
+    <ul class="menu">
+      <li>
+        <a href="index.html">Events</a>
+      </li>  
+      <li>
+        <a href="order.html">Orders</a>
+      </li>
+    </ul>
+  
+  </nav>
+  </header> 
+
+    <footer></footer>`
+}
+
+async function getAllEvents() {
+  const response = await fetch('http://localhost:7042/api/Event/GetAllEvents', {mode:'cors'});
+  const data = await response.json();
+  console.log(response);
+  console.log(data);
+  return data;
+}
+
+
+
+function displayCards(){
+  const container = document.querySelector('.card-container');
+  
+  const events = getAllEvents().then(events => events.forEach(element => {
+                const tempEvent = createEventElement(element)
+                container.appendChild(tempEvent);
+                }));
+  console.log(events);
+}
+
+
+function createEventElement(element){
+  console.log("Create element", element)
+  const newCard = document.createElement("div");
+  newCard.classList.add("card-container");
+  var imageUrl = ["https://play-lh.googleusercontent.com/ypVb0U7-YUPC3JqDyC9vEeeNNWxTxXVPeFZPLwMcVuUXrYFx2xJQxq3jBsyu8Dd1WQQ",
+  "https://electriccastle.ro/assets/img/ec-official-logo.jpg",
+  "https://3willowdesign.com/wp-content/uploads/2016/05/WineFest-logo-standard.jpg",
+  "https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcTvQwF25ly5E1xqdG7OwS7LoiW1rtiENfHRIkGCK-g&s"];
+
+  // var imageUrl =[];
+
+  if(element.name == "Untold"){
+    var image = imageUrl[0];
```

Comment:
Try to avoid as much as possible using `var` in js. `var` is not block scoped.  
Try to use `let` or `const`.